### PR TITLE
Specialize cached attention and cover fused transposed matmul tuning

### PR DESCRIPTION
**Pull request by kvark**
Follow-up to merged #190, based on `origin/main` (`8f05a139`).

- Compile cached attention at the graph's known head width through the existing per-head pipeline mechanism. No new kernel family or numerical shortcut.
- Include fused transposed matmuls in the existing qualified tile search, with transpose/addend CPU references. No new search budget or precision exception.
- Extend existing attention and tuning regressions; no new test executable.
- Document matched measurements and remaining gaps in `bench/gguf_latency.md`, without raw data or binaries.

**Review update (`52027a8`): reusable command buffers are removed.** The Blade pin is unchanged from main, and Blade #397 is closed. The replay source is retained separately at `experiment/serialized-replay-2026-09-19` (`6672ef3`) for an attribution mini-study. Replay's latency gains are no longer presented as production performance.

The diagnostic uses SmolLM2-135M, one sequence, 128-token prefill and batch-one cached decode, with full logits copied to the host. It exposes CPU recording/readback cost; it is not a batched-throughput study or the frozen Inferena cohort. Larger prefill blocks are not equivalent to concurrent decode requests.

Three reversed-order process pairs measured the fixed-head kernel change, with **ordinary command recording**:

| GPU | Phase | Before fixed head (ms) | Fixed head (ms) |
| --- | --- | ---: | ---: |
| RTX 5070 | Prefill | 11.58 | 8.82 |
| RTX 5070 | Decode/token | 4.29 | 3.20 |
| Arc B570 | Prefill | 27.63 | 22.79 |
| Arc B570 | Decode/token | 6.09 | 5.66 |

All 33 token predictions match an independent CPU f32 HF reference. Maximum per-row relative L2 remains below 0.000010 on NVIDIA and 0.000178 on Intel. CPU/GPU clocks were not fixed. Subsequent matched ordinary-recording controls were 8.65/3.13 ms on NVIDIA and 22.89/5.64 ms on Intel, versus llama.cpp 7.11/1.38 and 15.61/3.42 ms. **There is still a gap.** We accept ordinary recording as the production baseline and will continue improving GPU kernels.

This PR does not add attention split search. Stacked PR #193 now implements qualified, bounded search over the complete split/combine sequence at representative cache lengths. It selects different prefill/decode choices on NVIDIA and Intel automatically, with repeated measurements recorded there. Native-row layout probes remain separate experiments and must be remeasured without replay before production claims.

Verification after replay removal:

- All-target/all-feature Clippy and formatting passed.
- All 452 CPU unit tests passed (three ignored).
- The seven inference-op tests and shared-parameter regression passed on both NVIDIA and Intel.
- The known Naga Workgroup ArrayStride validation warning remains unchanged; no tolerance was relaxed.
- [CI run 736](https://github.com/kvark/meganeura/actions/runs/35465677712) passed all six jobs for this head.

The submitted paper and frozen benchmark cohort remain unchanged.

### CODE REVIEW

File: bench/gguf_latency.md [hunk 1]
```diff
@@ -160,6 +160,95 @@ A tilewise softmax-rescaling trial is retained at
...
+CPU reference on both devices; maximum per-row relative L2 error remains below
+0.000010 on NVIDIA and 0.000178 on Intel. The existing cache regression now also
+covers head widths 64, 80 and 512, including partially occupied thread lanes.
+NVIDIA attention pass intervals fell from 0.98 to 0.49 ms for decode and from
+3.80 to 1.66 ms for prefill in separate captures. This is still not parity with
+llama.cpp; matrix kernels and host submission remain substantial costs.
+
+### Serialized command replay
+
+The follow-up pins Blade `4befad4f5fbd427c1aca4b1b001fb8b7acd8e109` and reuses
```
Comment (kvark) [inline bench/gguf_latency.md]: I don't think we want serialized command/replay.  It's not idiomatic to Blade, and the wins are only visible on very small ML inference workloads, besides they are not even in GPU time. I think it's totally fine to say that our time on this benchmark is hitting this limitation of command recording that we find acceptable, since the limitation is not going to affect us in practice.
Caveat - high-frequency trading would probably want that, but that's not the goal right now.

Comment (kvark) [reply bench/gguf_latency.md]: Removed the replay integration and its invalidation state in `52027a8`; the Blade dependency now matches main again. Blade #397 is closed. The exact experiment remains on a separate source-only branch, and this section labels its timings as a mini-study rather than production results.

The diagnostic is one sequence, 128-token prefill and one-token decode. That is a valid interactive-latency case but too narrow to drive the general runtime design. I corrected the PR's numbers to ordinary-recording performance. Attention split counts are a separate, currently missing tuner axis: we should measure and qualify complete split/combine sequences for the shape/device and varying cache lengths, not install the pilot's winning count as a new threshold.
